Update and add new model listing options

helderlopes97 opened the pull request.

helderlopes97 committed:
feat: update and add new model listing options

helderlopes97 committed:
feat: add reaction completion to the models listed

C-nit commented on README.md:
````diff
@@ -59,6 +59,40 @@ rxn4chemistry_wrapper.create_project('test_wrapper')
+response = rxn4chemistry_wrapper.list_models_by_scope(scope="REACTIONPROPERTYPREDICTOR", category_name="atom-mapping", project_id="<project_id>")
+print(response)
+```
+
+To list model categories by scope:
+
+```python
+# all model categories for the REACTIONPROPERTYPREDICTOR scope
+response = rxn4chemistry_wrapper.list_models_categories_by_scope(scope="REACTIONPROPERTYPREDICTOR")
+print(response)
+```
````
1. This feels like a duplication, maybe the first example can be some other scope like retro synthesis, something without category.

2. Next to showing the use of the property predictor with model category, it should be pointed out that for each category there is also it's on scope version, e.g. `ATOMMAPPING`

helderlopes97 committed:
feat: added options to use the new dedicated endpoints for atom mapping, yield and fingerprint predictions
- organized all urls and added new ones
- added list_all_models (for admin users)
- empy response when using invalid scope in list_models_by_scope
- renamed predict_reaction_properties to predict_reaction_properties_from_smiles
- added predict_reaction_properties_from_file
- added atom_mapping_from_smiles
- added atom_mapping_from_file
- added yield_from_smiles
- added yield_from_file
- added fingerprint_from_smiles
- added fingerprint_from_file
- run black and isort

C-nit committed:
Merge branch 'main' into update-wrapper

C-nit reviewed (approved):
Approved

C-nit committed:
chore: bump version to 1.15.0
